src/firefly/js/ui/DownloadDialog.jsx

```diff
@@ -119,16 +123,24 @@ let dlTitleIdx = 0;
 const newBgKey = () => 'DownloadOptionPanel-' + Date.now();
 
 export function DownloadOptionPanel (props) {
-    const {groupKey, cutoutSize, help_id, children, style, title, tbl_id, dlParams, dataTag} = props;
+    const {groupKey, cutoutSize, help_id, children, style, title, tbl_id, dlParams,imageSource, dataTag} = props;
     const { showZipStructure=true, showEmailNotify=true, showFileLocation=true, showTitle=true } = props;
 
     const labelWidth = 110;
     const ttl = title || DOWNLOAD_DIALOG_ID;
     const [bgKey, setBgKey] = useState(newBgKey());
 
     const onSubmit = useCallback((formInputs={}) => {
-        var {request, selectInfo} = getTblById(tbl_id);
-        const {FileGroupProcessor} = dlParams;
+
+        let {request,  selectInfo} = getTblById(tbl_id);
+        const {FileGroupProcessor, SearchProcessor} = dlParams;
+        if (SearchProcessor){
+            request.id = SearchProcessor;
+        }
+        if (imageSource){
+            request.selectImage = imageSource;
+        }
+        
```

Comment:
yes, i think this has to be passed from the parent, no need to have distinction here in Firefly code. If i understood Loi correctly, you need to add a function as a prop to the dialog and pass it from outiside, correct?

Reply:
@Loi 
Can you please check the function prop added in DownloadDialog?  The function is passed from FinderchartDownloadDialog.  Every component who is using DownloadDialog can pass its own function if it is needed.  Thank you for your good advice.  Using the function property, it is easy to fix the bug you noted in multi-search.   

@ejoliet The three color issue is from the old dev.  Since I did not update my pull request, there was an exception in PlotImageTask when the 3-color is requested.   I will test it after I rebase the dev.

Reply:
@loitly @ejoliet 
Please only check the function properties if you wish.   I fixed the issue for current target.  
I rebased with dev and am testing/debugging color image and multi-position search for all target.  The color image issue is not related to what I am doing.   It is uploading image failure.  Once I have these two issues solved, I will build the test URL.  FYI